lib/buffer.js

```diff
@@ -300,8 +300,28 @@ function byteLength(string, encoding) {
 
 Buffer.byteLength = byteLength;
 
+
+Buffer.prototype.copy = function(target, targetStart, sourceStart, sourceEnd) {
+  if (!(this instanceof Buffer))
+    throw new TypeError('this is not an instance of Buffer');
+
+  if (!(target instanceof Buffer))
+    throw new TypeError('target is not an instance of Buffer');
+
+  return binding.copy(this, target, targetStart, sourceStart, sourceEnd);
+};
+
 // toString(encoding, start=0, end=buffer.length)
 Buffer.prototype.toString = function(encoding, start, end) {
+  if (!(this instanceof Buffer)) {
+    // If no arguments, then this is probably an implicit coercion, and we
+    // shouldn't throw.
+    if (arguments.length === 0)
+      return Object.prototype.toString.call(this);
+    else
+      throw new TypeError('this is not an instance of Buffer');
```

Comment:
Even though this case is insane, it inevitably will break someone in the wild. Because of that this is a sem-minor patch.